src/components/CreateNimi/partials/NimiLinkField/NimiLinkField.tsx

```diff
@@ -14,27 +15,27 @@ export interface NimiLinkFieldProps {
  */
 export function NimiLinkField({ link, label }: NimiLinkFieldProps) {
   const { setValue, getValues } = useFormContext<Nimi>();
+  const [inputValue, setInputValue] = useState('');
 
   const onChange: ChangeEventHandler<HTMLInputElement> = (event) => {
-    const prevState = getValues('links') || [];
-    const hasLink = prevState.some((prevLink) => prevLink.type === link);
-    const newState: NimiLinkBaseDetails[] = hasLink
-      ? prevState.map((curr) => {
-          if (curr.type === link) {
-            return { ...curr, url: event.target.value };
-          }
+    setInputValue(event.target.value);
```

Comment:
We've removed the entire logic. I might be missing something here

Reply:
@adamazad This pr is work in progress still...But I put that logic inside the replaceOrAddArrayItem

Reply:
Im just having trouble with it since the object is a bit different for links and addresses...